docs/azure/integrations-overview.md

```diff
@@ -40,6 +64,116 @@ The preceding code:
+    .WithQueue("queue");
+```
+
+The preceding code:
+
+- Creates a new `builder` instance.
+- Adds a parameter named `existingResourceName` to the builder.
+- Adds an Azure Service Bus resource named `messaging` to the builder.
+- Calls the `RunAsExisting` method on the `serviceBus` resource builder, passing the `existingResourceName` parameter.
+- Adds a queue named `queue` to the `serviceBus` resource.
+
+### Express existing Azure resources for publish mode
+
+The `PublishAsExisting` method is used in "publish" mode when the intent is to declare and reference an already-existing Azure resource during publish mode. This API facilitates the creation of manifests and templates that include resource definitions that map to existing resources in Bicep.
+
+To mark an Azure resource as existing in for the "publish" mode, call the `PublishAsExisting` method on the resource builder. Consider the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder();
+
+var existingResourceName = builder.AddParameter("existingResourceName");
+
+var serviceBus = builder.AddAzureServiceBus("messaging")
+    .PublishAsExisting(existingResourceName)
+    .WithQueue("queue");
+```
+
+The preceding code:
+
+- Creates a new `builder` instance.
+- Adds a parameter named `existingResourceName` to the builder.
+- Adds an Azure Service Bus resource named `messaging` to the builder.
+- Calls the `PublishAsExisting` method on the `serviceBus` resource builder, passing the `existingResourceName` parameter.
+- Adds a queue named `queue` to the `serviceBus` resource.
+
+After the app host is executed in publish mode, the generated manifest file will include the `existingResourceName` parameter, which can be used to reference the existing Azure resource. Consider the following generated manifest file:
+
+```json
+{
+  "type": "azure.bicep.v0",
+  "connectionString": "{messaging.outputs.serviceBusEndpoint}",
+  "path": "messaging.module.bicep",
+  "params": {
+    "existingResourceName": "{existingResourceName.value}",
+    "principalType": "",
+    "principalId": ""
+  }
+}
+```
+
+For more information on the manifest file, see [.NET Aspire manifest format for deployment tool builders](../deployment/manifest-format.md).
+
+Additionally, the generated Bicep template will include the `existingResourceName` parameter, which can be used to reference the existing Azure resource. Consider the following generated Bicep template:
+
+```bicep
+@description('The location for the resource(s) to be deployed.')
+param location string = resourceGroup().location
+
+param existingResourceName string
+param principalType string
+param principalId string
+
+resource messaging 'Microsoft.ServiceBus/namespaces@2024-01-01' existing = {
+  name: existingResourceName
+}
+
+output serviceBusEndpoint string = messaging.properties.serviceBusEndpoint
+```
+
+For more information on the generated Bicep template, see [Infrastructure as code](#infrastructure-as-code).
```

Comment:
@eerhardt discovered some APIs that are incompatible with the existing resource model. For example, you can't call `UseAccessKeyAuthentication` on an existing Azure Redis resource or `UsePasswordAuthentication` on an existing Postgres resource. We might want to add a warning about that here.

Reply:
You can call these methods, but they just need to "match up" with what is already enabled on the server. For example `UsePasswordAuthentication` on an existing resource works if you pass the username/password in that is already configured on the server.